packages/@lwc/engine-core/src/framework/rendering.ts

```diff
@@ -231,6 +232,7 @@ function mountStatic(
     const elm = (vnode.elm = cloneNode(vnode.fragment, true));
 
     linkNodeToShadow(elm, owner, renderer);
+    applyElementRestrictions(elm, vnode);
```

Comment:
This looks like a functional change,  since we wren't doing anything here before.

IIUC, we weren't applying restrictions to synthetic shadow elements that use the static optimization. So is the current `master` code broken?